src/Disks/DiskLocal.cpp

```
@@ -319,9 +319,21 @@ DiskDirectoryIteratorPtr DiskLocal::iterateDirectory(const String & path)
 {
     fs::path meta_path = fs::path(disk_path) / path;
     if (!broken && fs::exists(meta_path) && fs::is_directory(meta_path))
-        return std::make_unique<DiskLocalDirectoryIterator>(disk_path, path);
-    else
-        return std::make_unique<DiskLocalDirectoryIterator>();
+    {
+        try
+        {
+            fs::directory_iterator entry(fs::path(disk_path) / path);
+            return std::make_unique<DiskLocalDirectoryIterator>(std::move(entry), path);
+        }
+        catch (const fs::filesystem_error & e)
+        {
+            if (e.code() == std::errc::no_such_file_or_directory)
+                return std::make_unique<DiskLocalDirectoryIterator>();
+            throw;
```

I'm not sure that it's good idea to ignore the exception in general case.